tests/Feature/Api/PublicScheduleTest.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+namespace Tests\Feature\Api;
+
+use App\Models\Lecturing;
+use Carbon\Carbon;
+use Illuminate\Foundation\Testing\RefreshDatabase;
+use Tests\TestCase;
+
+class PublicScheduleTest extends TestCase
+{
+    use RefreshDatabase;
+
+    /** @test */
+    public function can_get_schedules_within_specified_date_range() 
+    {
+        $lecturing = factory(Lecturing::class)->create([
+            'audience_code' => Lecturing::AUDIENCE_FRIDAY,
+            'date' => Carbon::tomorrow()->format('Y-m-d'),
+        ]);
+
+        $unlistedLecturing = factory(Lecturing::class)->create([
+            'audience_code' => Lecturing::AUDIENCE_PUBLIC,
+            'date' => Carbon::yesterday()->format('Y-m-d'),
+        ]);
```

Comment:
Ini juga perlu kita bedakan nama petugas nya, karena di paling bawah kita membandingkan nama-nama petugas (tampil di json atau tidak).

```suggestion
        $lecturing = factory(Lecturing::class)->create([
            'audience_code' => Lecturing::AUDIENCE_FRIDAY,
            'date' => Carbon::tomorrow()->format('Y-m-d'),
            'lecturer_name' => 'Ustadz Haikal',
            'imam_name' => 'Ustadz Febri',
            'muadzin_name' => 'Ahmad',
        ]);

        $unlistedLecturing = factory(Lecturing::class)->create([
            'audience_code' => Lecturing::AUDIENCE_PUBLIC,
            'date' => Carbon::yesterday()->format('Y-m-d'),
            'lecturer_name' => 'Ustadz Fauzan',
            'imam_name' => 'Abdussamad',
            'muadzin_name' => 'Tono',
        ]);
```